instrumentation/jetty-httpclient/jetty-httpclient-9.2/testing/src/main/groovy/io/opentelemetry/javaagent/instrumentation/jetty/httpclient/v9_2/AbstractJettyClient9Test.groovy

```diff
@@ -0,0 +1,156 @@
+import org.eclipse.jetty.client.HttpClient
+import org.eclipse.jetty.client.api.ContentResponse
+import org.eclipse.jetty.client.api.Request
+import org.eclipse.jetty.client.api.Response
+import org.eclipse.jetty.client.api.Result
+import org.eclipse.jetty.http.HttpMethod
+import org.eclipse.jetty.util.ssl.SslContextFactory
+import org.junit.Rule
+import org.junit.rules.TestName
+import spock.lang.Shared
+
+import java.util.concurrent.TimeUnit
+
+abstract class AbstractJettyClient9Test extends HttpClientTest<Request> {
+
+  abstract HttpClient createStandardClient()
+
+  abstract HttpClient createHttpsClient(SslContextFactory sslContextFactory)
+
+
+  @Shared
+  def client = createStandardClient()
+  @Shared
+  def httpsClient = null
+
+  @Rule
+  TestName name = new TestName()
+
+  Request jettyRequest = null
+
+  def setupSpec() {
+
+    //Start the main Jetty HttpClient
+    client.start()
+
+    SslContextFactory tlsCtx = new SslContextFactory()
+//    tlsCtx.setExcludeProtocols("TLSv1.3")
+    httpsClient = createHttpsClient(tlsCtx)
+    httpsClient.setFollowRedirects(false)
+    httpsClient.start()
+  }
+
+  @Override
+  Request buildRequest(String method, URI uri, Map<String, String> headers) {
+
+    HttpClient theClient = uri.scheme == 'https' ? httpsClient : client
+
+    Request request = theClient.newRequest(uri)
+
+    HttpMethod methodObj = HttpMethod.valueOf(method)
+    request.method(methodObj)
+    request.timeout(CONNECT_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+
+    jettyRequest = request
+
+    return request
+  }
+
+  @Override
+  String userAgent() {
+    if (name.methodName.startsWith('connection error') && jettyRequest.getAgent() == null) {
+      return null
+    }
+    return "Jetty"
+  }
+
+  @Override
+  int sendRequest(Request request, String method, URI uri, Map<String, String> headers) {
+    headers.each { k, v ->
+      request.header(k, v)
+    }
+
+    ContentResponse response = request.send()
+
+    return response.status
+  }
+
+  private static class JettyClientListener implements Request.FailureListener, Response.FailureListener {
+
+    volatile Throwable failure
+
+    @Override
+    void onFailure(Request requestF, Throwable failure) {
+      this.failure = failure
+
+    }
+
+    @Override
+    void onFailure(Response responseF, Throwable failure) {
+      this.failure = failure
+    }
+
+  }
+
+  @Override
+  void sendRequestWithCallback(Request request, String method, URI uri, Map<String, String> headers, RequestResult requestResult) {
+
+    JettyClientListener jcl = new JettyClientListener()
+
+    request.onRequestFailure(jcl)
+    request.onResponseFailure(jcl)
+    headers.each { k, v ->
+      request.header(k, v)
+    }
+    Context parentContext = Context.current()
+    Scope scope = parentContext.makeCurrent()
+//    attachInterceptor(request, parentContext)
+
+    request.send(new Response.CompleteListener() {
+      @Override
+      void onComplete(Result result) {
```

Comment:
`Result` seems to have failure, don't think we need the `jcl`

https://www.eclipse.org/jetty/javadoc/jetty-9/org/eclipse/jetty/client/api/Result.html

Reply:
Nope, have to have the extra protection, Jetty client doesn't do a good job setting the failure field on their Result objects.   callback unit test start failing if I try to rely only on Result object